crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -437,6 +483,41 @@ impl<T: NodeStorage> MerkleTree<T> {
         Ok(result)
     }
 
+    /// Generates a merkle-proof a given `key`.
```

Comment:
Not sure if you meant "given a key" or "for a given key"

```suggestion
    /// Generates a merkle-proof for a given `key`.
```